staging/src/k8s.io/apiserver/pkg/server/healthz/healthz_test.go

```diff
@@ -82,6 +85,9 @@ func testMultipleChecks(path string, t *testing.T) {
 		addBadCheck      bool
 	}{
 		{"?verbose", "[+]ping ok\nhealthz check passed\n", http.StatusOK, false},
+		{"?exclude=dontexist", "ok", http.StatusOK, false},
```

Comment:
Is this the right thing to do? How do we help people not make configuration errors?

Reply:
I initially opted to do nothing but was not sure if this was the correct thing to do, so I added the test to explicitly draw out this behavior. As it appears to me, we have three options:

1. do nothing
2. log an error
3. throw an error (possibly a 404 or 500)

Thinking about it some more, we should at least log something, if not opt for an error. Do you have a particular preference? I don't think this will be a common user interaction with kubernetes, since it involves modification of the manifest file for the kube-apiserver, which I imagine will only mostly only be necessary on the cloud-provider level.

Reply:
I'm tempted to say to throw an error, and the only reason I didn't suggest it is the case where we remove a health check that could currently be in the config files.

We could add in the code a list of removed health checks + a test exercising all existing ones to ensure it stays up to date?

Reply:
Maybe @deads2k has an opinion.

Reply:
I'm actually okay with throwing an error. Removing a health check wholesale is pretty backwards incompatible, so I would think that a change like that would be almost impossible to get in without breaking lots of things. If we wanted a change like that, I think it'd be better to version the healthz endpoint (v2), or have other people exclude the removed health check in their cloud provider manifests themselves.

Reply:
Maybe we just `glog.Warningf` from apiserver if we see an unknown check being excluded - same path as a deprecated flag?

Reply:
Sorry, was adding the error behavior when i caught this bug [#70752](https://github.com/kubernetes/kubernetes/issues/70752). I've fixed it in a separate PR [70753](https://github.com/kubernetes/kubernetes/pull/70753). I'll wait to merge that one before proceeding here.

Reply:
> I'm tempted to say to throw an error, and the only reason I didn't suggest it is the case where we remove a health check that could currently be in the config files.

Throwing an error makes it harder to skew across versions and support rollback without modifying config/deployment logic.

Reply:
@deads2k, is it common to rollback versions without also changing the manifest file?